### PR TITLE
Add unit tests for some tracks in product editor

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Partially addresses https://github.com/woocommerce/woocommerce/issues/48142 for the following tracks:

1. wcadmin_product_add_view
2. wcadmin_product_edit_view
1. wcadmin_product_add_publish
1. wcadmin_product_tab_click
1. wcadmin_product_editor_feedback_bar_turnoff_editor_click
1. wcadmin_product_dropdown_option_click

### How to test the changes in this Pull Request:

1. Check test codes
3. Ensure tests passes

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: packages/js/internal-js-tests/jest-preset.js [hunk 1]
```diff
@@ -51,6 +51,9 @@ module.exports = {
 		'**/test/*.[jt]s?(x)',
 		'**/?(*.)test.[jt]s?(x)',
 	],
+	testPathIgnorePatterns: [
```
Comment (ilyasfoo) [inline packages/js/internal-js-tests/jest-preset.js:54]: `global.d.ts` was being run by jest since I've introduced one in test folder in this PR

**Review by nathanss — APPROVED** [review]
Nice job @ilyasfoo !

I've reviewed and ran the tests locally, they are passing and are a nice addition to our codebase. Thanks!

File: packages/js/product-editor/src/components/feedback-bar/test/feedback-bar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+jest.mock( '@woocommerce/tracks', () => ( {
+	...jest.requireActual( '@woocommerce/tracks' ),
+	recordEvent: jest.fn(),
+} ) );
+
+describe( 'FeedbackBar', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+	it( 'should render empty name rows when names are empty', () => {
```
Comment (chihsuan) [inline packages/js/product-editor/src/components/feedback-bar/test/feedback-bar.test.tsx]: The test case name mentions rendering empty name rows, but the test is about turning off the feedback bar. Is this expected?

**Review by chihsuan — APPROVED** [review]
LGTM. 👍   Just left a comment about the test case name.

File: packages/js/product-editor/src/components/feedback-bar/test/feedback-bar.test.tsx (hunk 2 above)
Comment (ilyasfoo) [reply packages/js/product-editor/src/components/feedback-bar/test/feedback-bar.test.tsx]: Good catch! Copy pasta error. Fixed in aef7ae2

**Pull request merged**